2_process/src/process_nhdv2_attr.R

```diff
@@ -269,3 +269,184 @@ process_catchment_nhdv2_attr <- function(file_path,vars_table,segs_w_comids,nhd_
+    #RUN7100 seems like it is by HUC02 instead of reach.
+    select(!contains(drop_columns)) %>%
+    #Modify the CAT Basin Areas that are 0 with PRMS areas
+    #These areas are otherwise nearly identical (max difference of 0.1 sq.km)
+    mutate(CAT_BASIN_AREA_sum = case_when(CAT_BASIN_AREA_sum == 0 ~ AREASQKM_PRMS,
+                                          TRUE ~ CAT_BASIN_AREA_sum)) %>%
+    #drop PRMS area column
+    select(-AREASQKM_PRMS)
+  
+  #RECHG
+  #Change recharge for NA segment to the average of its neighbors (from_segs and to_seg)
+  #index to change
+  nhdv2_attr_refined$CAT_RECHG_area_wtd <- refine_from_neighbors(nhdv2_attr_refined,
+                                                                 attr_i = 'CAT_RECHG_area_wtd', 
+                                                                 prms_reach_attr)
+  
+  #EWT - water table
+  #Change EWT for segment with deep value to the average of its neighbors (from_segs and to_seg)
+  nhdv2_attr_refined$CAT_EWT_area_wtd <- refine_from_neighbors(nhdv2_attr_refined,
+                                                               attr_i = 'CAT_EWT_area_wtd',
+                                                               prms_reach_attr)
+  
+  #STRM_DENS
+  #Compute stream density from the NHD catchment reach length and area
+  #only for the 5 NA PRMS segments. These have 1 or 2 NHD catchments.
+  # other PRMS segments with some NA stream densities cover areas <3% of total area.
+  #Gather the PRMS areas for these reaches
+  ind_areas <- filter(nhdv2_attr_refined, is.na(CAT_STRM_DENS_area_wtd)) %>%
+    select(PRMS_segid, CAT_BASIN_AREA_sum)
+  #Gather the sum of NHD reach lengths in m
+  ind_areas$length_m <- 0
+  for (i in 1:nrow(ind_areas)){
+    #all NHD reaches for this PRMS segment
+    nhd_reaches <- filter(prms_nhdv2_xwalk,
+                          PRMS_segid %in% ind_areas$PRMS_segid[i]) %>%
+      select(comid_cat) %>%
+      str_split(., pattern = ';', simplify = T)
+    
+    ind_areas$length_m[i] <- filter(nhdv2_reaches,
+                                    COMID %in% nhd_reaches) %>%
+      select(LENGTHKM) %>% st_drop_geometry() %>%
+      sum()
+  }
+  #Compute the reach stream density length (km)/area (sq.km)
+  #There must be a typo in the table's units because using m length gives
+  #results that are 3 orders of magnitude larger than other values
+  ind_areas <- mutate(ind_areas, str_dens = length_m/CAT_BASIN_AREA_sum) %>%
+    select(-length_m, -CAT_BASIN_AREA_sum)
+  #assign to attribute table
+  nhdv2_attr_refined <- mutate(nhdv2_attr_refined,
+                          CAT_STRM_DENS_area_wtd = case_when(PRMS_segid %in% ind_areas$PRMS_segid ~
+                                                               ind_areas$str_dens[match(PRMS_segid, ind_areas$PRMS_segid)],
+                                                             TRUE ~ CAT_STRM_DENS_area_wtd)
+  )
+  
+  #Compute TOT variables from PRMS CAT variables
```

Comment:
Do you mind adding a comment that explains why it's necessary to compute TOT variables from CAT for these attributes? That might be helpful for interpreting this code at some later date.